File: problems/Easy/1051. Height Checker.cpp
class Solution {
public:
    int heightChecker(vector<int>& heights) {
        vector<int> tmp=heights;
        sort(heights.begin(), heights.end());
        int res=0, n=heights.size();
        for(int i=0;i<n;++i){
            if(heights[i]!=tmp[i]){
                res++;
            }
        }
        return res;
    }
};

File: problems/Easy/1051. Height Checker.go
func heightChecker(heights []int) int {
    // fmt.Println(heights)
    expected := make([]int, len(heights))
    copy(expected, heights)
    res := 0
    slices.Sort(expected)
    // fmt.Println(expected)
    for i, height := range heights {
        // fmt.Println(height, expected[i])
        if height != expected[i] {
            res++
        }
    }
    return res
}